src/main/java/org/commonjava/util/partyline/impl/infinispan/model/InfinispanJFS.java

```diff
@@ -168,11 +185,25 @@ public void onCacheCreatedEvent( CacheEntryEvent event )
             if ( event.getKey() == this.key )
             {
                 // TODO: Do something here to notify
-                return;
+                this.lock.signal();
```

Comment:
I'd also set a flag internally on this listener. Then, if in future some code called listenToCache() with an already-triggered listener, you could just return immediately.